tests/devices/test_default_qubit_broadcasting.py

```diff
@@ -0,0 +1,2948 @@
+            [qml.QubitStateVector(np.array(input), wires=[0])], obs.diagonalizing_gates()
+        )
+        res = qubit_device_1_wire.var(obs)
+
+        assert np.allclose(res, expected_output, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize(
+        "operation,input,expected_output,par",
+        [
+            (qml.Hermitian, [[1, 0], [0, 1], [INVSQ2, -INVSQ2]], [1, 1, 1], [[1, 1j], [-1j, 1]]),
+        ],
+    )
+    def test_var_single_wire_with_parameters_broadcasted(
+        self, qubit_device_1_wire, tol, operation, input, expected_output, par
+    ):
+        """Tests that variances are properly calculated for single-wire observables with parameters."""
+
+        obs = operation(np.array(par), wires=[0])
+
+        qubit_device_1_wire.reset()
+        qubit_device_1_wire.apply(
+            [qml.QubitStateVector(np.array(input), wires=[0])], obs.diagonalizing_gates()
+        )
+        res = qubit_device_1_wire.var(obs)
+
+        assert np.allclose(res, expected_output, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize(
+        "operation,input,expected_output,par",
+        [
+            (
+                qml.Hermitian,
+                [
+                    [1 / math.sqrt(3), 0, 1 / math.sqrt(3), 1 / math.sqrt(3)],
+                    [0, 0, 0, 1],
+                    [1 / math.sqrt(2), 0, -1 / math.sqrt(2), 0],
+                ],
+                [11 / 9, 2, 3 / 2],
+                [[1, 1j, 0, 1], [-1j, 1, 0, 0], [0, 0, 1, -1j], [1, 0, 1j, 1]],
+            ),
+            (
+                qml.Hermitian,
+                [[INVSQ2, 0, 0, INVSQ2], [0, INVSQ2, -INVSQ2, 0]],
+                [0, 0],
+                [[1, 0, 0, 0], [0, -1, 0, 0], [0, 0, -1, 0], [0, 0, 0, 1]],
+            ),
+        ],
+    )
+    def test_var_two_wires_with_parameters_broadcasted(
+        self, qubit_device_2_wires, tol, operation, input, expected_output, par
+    ):
+        """Tests that variances are properly calculated for two-wire observables with parameters."""
+
+        obs = operation(np.array(par), wires=[0, 1])
+
+        qubit_device_2_wires.reset()
+        qubit_device_2_wires.apply(
+            [qml.QubitStateVector(np.array(input), wires=[0, 1])], obs.diagonalizing_gates()
+        )
+        res = qubit_device_2_wires.var(obs)
+
+        assert np.allclose(res, expected_output, atol=tol, rtol=0)
+
+    def test_var_estimate_broadcasted(self):
+        """Test that the variance is not analytically calculated"""
+
+        dev = qml.device("default.qubit", wires=1, shots=3)
+
+        @qml.qnode(dev, diff_method="parameter-shift")
+        def circuit():
+            qml.RX(np.zeros(5), wires=0)  # Broadcast the tape without applying an op
+            return qml.var(qml.PauliX(0))
+
+        var = circuit()
+
+        # With 3 samples we are guaranteed to see a difference between
+        # an estimated variance and an analytically calculated one
+        assert np.all(var != 1.0)
+
+
+class TestSampleBroadcasted:
```

Comment:
How about broadcasting `state` or `probs`? I imagine if we have some parameterized gates that are applied, then broadcasting the parameters would result in various state vectors and probs

Reply:
Fair enough! :sweat_smile: I am not sure the original `test_default_qubit.py` has any? :thinking:

Reply:
@josh146 Do you know whether these tests are somewhere else/are not needed?

Reply:
Since `default.qubit` is so old, the tests are a bit of a mess :/

- The actual `test_default_qubit.py` file doesn't really have unit tests, and is instead mostly integration tests. So the high coverage of `default.qubit` is only due to every single other test in the test suite using `default.qubit` lol

- We do have state tests in `test_measurements.py`, but they are again integration tests that use `default.qubit`, so a bit of a mess